test/scripts/boot-image

```diff
@@ -156,6 +156,15 @@ def ensure_can_run_qemu_test(arch, image_path, config_file):
     customizations = config.get("blueprint", {}).get("customizations", {})
     if "openscap" in customizations:
         raise CannotRunQemuTest("openscap is not working right now")
+    # network-installer has a bunch of extra constraints
+    if config.get("image-type") in ["network-installer", "everything-network-installer"]:
+        # network install needs subscribed content but we cannot provide
+        # that currently
+        match config.get("distro"):
+            case name if name.startswith("rhel"):
+                raise CannotRunQemuTest("rhel network-installer needs subscription keys which we cannot provide yet")
+            case "centos-10":
```

Comment:
I strongly suspect we need to also skip centos-9 because afaict it will not automatically use a network repo and in our unattended kickstart file we do not include a "url" - this seems to be handled automatically (and correctly) in centos-10s anaconda.

Reply:
There is now also a `software selection error` in fedora 43 (c.f. https://gitlab.com/redhat/services/products/image-builder/ci/images/-/jobs/12121649535) which might (or might not be) a fluke